pilot/pkg/model/telemetry.go

```diff
@@ -1052,3 +1053,51 @@ func generateStatsConfig(class networking.ListenerClass, metricsCfg telemetryFil
 func disableHostHeaderFallback(class networking.ListenerClass) bool {
 	return class == networking.ListenerClassSidecarInbound || class == networking.ListenerClassGateway
 }
+
+type OTelResourceSemantic string
+
+const (
+	KubernetesCluster   OTelResourceSemantic = "k8s.cluster.name"
+	KubernetesNamespace OTelResourceSemantic = "k8s.namespace.name"
+	KubernetesPod       OTelResourceSemantic = "k8s.pod.name"
+	ServiceName         OTelResourceSemantic = "service.name"
+)
+
+// resourceAttributes return semantic attributes follow OpenTelemetry specification
+// see more details here: https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/
+func resourceAttributes(proxy *Proxy) *otlpcommon.KeyValueList {
+	// TODO(zirain): generate resource attributes for VM
+	if proxy.IsVM() {
+		return nil
+	}
+
+	podName, namespace := extractProxyID(proxy)
+
+	return &otlpcommon.KeyValueList{
+		Values: []*otlpcommon.KeyValue{
+			otelKeyValue(KubernetesCluster, proxy.Metadata.ClusterID.String()),
+			otelKeyValue(KubernetesNamespace, namespace),
```

Comment:
```suggestion
			otelKeyValue(KubernetesNamespace, proxy.ConfigNamespace),
```

Reply:
As asked a few times, please share this logic with the baggage setup.

Reply:
do you means share generate logic or someting else?

Reply:
Share the same logic and code